pkg/cnab/provider/parameters_test.go

```diff
@@ -230,3 +282,181 @@ func Test_loadParameters_fileParameter(t *testing.T) {
+					false, false, false, true, nil, "",
+				},
+				{"not required, not provided, default does not exist, does not apply to action",
+					false, false, false, false, nil, "",
+				},
+			}
+
+			for _, tc := range testcases {
+				t.Run(tc.name, func(t *testing.T) {
+					c := config.NewTestConfig(t)
+					instanceStorage := instancestorage.NewTestInstanceStorageProvider()
+					d := NewRuntime(c.Config, instanceStorage)
+
+					bun := &bundle.Bundle{
+						Name:          "mybuns",
+						Version:       "v1.0.0",
+						SchemaVersion: "v1.0.0",
+						Actions: map[string]bundle.Action{
+							"zombies": {
+								Modifies: true,
+							},
+						},
+						InvocationImages: []bundle.InvocationImage{
+							{
+								BaseImage: bundle.BaseImage{
+									Image:     "mybuns:latest",
+									ImageType: "docker",
+								},
+							},
+						},
+						Definitions: definition.Definitions{
+							"my-param": &definition.Schema{
+								Type: "string",
+							},
+						},
+						Parameters: map[string]bundle.Parameter{
+							"my-param": bundle.Parameter{
+								Definition: "my-param",
+								Required:   tc.required,
+							},
+						},
+					}
+
+					if tc.defaultExists {
+						bun.Definitions["my-param"].Default = "my-param-default"
+					}
+
+					if !tc.appliesToAction {
+						param := bun.Parameters["my-param"]
+						param.ApplyTo = []string{"non-applicable-action"}
+						bun.Parameters["my-param"] = param
+					}
+
+					args := ActionArguments{
+						Claim:    "test",
+						Insecure: true,
+						Driver:   "debug",
+					}
+					// If param is provided (via --param/--param-file)
+					// it will be attached to args
+					if tc.provided {
+						args.Params = map[string]string{
+							"my-param": "my-param-value",
+						}
+					}
+
+					// If action is install, no claim is expected to exist
+					// so we write a bundle and pull it in via ActionArguments
+					if action == "install" {
+						bytes, err := json.Marshal(bun)
+						require.NoError(t, err)
+
+						// We currently need to read/write from the same file on disk
+						// as cnab-go's bundle loader still makes raw os calls for loading a bundle
+						err = ioutil.WriteFile("testdata/bundle.json", bytes, 0644)
+						require.NoError(t, err)
+
+						args.BundlePath = "testdata/bundle.json"
+					} else {
+						// For all other actions, a claim is expected to exist
+						// so we create one here and add the bundle to the claim
+						claim, err := claim.New("test")
+						require.NoError(t, err)
+
+						claim.Bundle = bun
+						d.instanceStorage.Store(*claim)
+					}
+
+					var err error
+					switch action {
+					case "install":
+						err = d.Install(args)
+					case "upgrade":
+						err = d.Upgrade(args)
+					case "invoke":
+						err = d.Invoke("zombies", args)
+					case "uninstall":
+						err = d.Uninstall(args)
+					}
+
+					if tc.expectedErr != "" {
+						require.EqualError(t, err, tc.expectedErr)
+					} else if action != "uninstall" {
```

Comment:
I don't understand why this `if action != "uninstall"` is here instead of only a bare `else`. Where are the asserts happening for the uninstall case? We should at least check that uninstall didn't return an error right? Even if the claim was deleted?

Reply:
Ah, indeed, it currently has that gap of checking the uninstall case (that no err occurred when not expected).  Thank you; updating!